coda.io.subdomain.json

```diff
@@ -0,0 +1,25 @@
+{  
+    "providerId":"coda.io",
+    "providerName":"Coda",
+    "serviceId":"subdomain",
+    "serviceName":"Coda Site",
+    "version": 1,
+    "syncPubKeyDomain" : "coda.io",
+    "logoUrl":"https://cdn.coda.io/icons/png/color/coda-192.png",
+    "description":"Allows serving Coda documents from your own domain (subdomain only)",
+    "variableDescription":"`subdomain` refers to the subdomain that the user wishes to use (not just limited to 'www'). `environment` refers to the environment for Coda's custom domain server. `docId` refers to the coda document used for custom domain. `verification` refers to the TXT verification of doc ownership",
+    "records":[  
+       {  
+          "type":"CNAME",
+          "host":"%subdomain%",
```

Comment:
Looking at the spec, it looks like host is used for a static string. What if the host is up to the discretion of the user?

for ex: Say someone wants to setup either a custom domain `www.example.com` as well as `abc.example.com`

We at Coda would like to be able to support someone's ability to choose whatever host they choose, rather than sticking to just `www`.

I'm looking at the extenuating circumstances for adding variables for the `host` parameter. This `host` is only really set for TXT and for CNAME since we wanted to respect any host for CNAMEs only (A record we only use for APEX publishing, so there is no host). If this is the case, should I just leave `subdomain` and pass that as a query? Our TXT record also should respect the custom host that a user passes in, which is why I have it as `_coda.%subdomain%`. Is there another way to honor this requirement for the template?

Reply:
@johnli-codaio if you place the configuration `"host": "@"` in CNAME and `"host"="_coda"` for TXT and then when applying the template you'd use `host=www` or 'host=abc` query parameter, the DNS provider will create records with the desired subdomain. This is the purpose of this reserved `host` parameter to do exactly that.
If you mimic the same with own variable, you will run into trouble if you ever try to add the templates to more than one subdomain of the same domain. Some DNS providers take domain+host pair as unique instance key, so any successive apply will be treated as an update and other records (you other subdomain) deleted.

Reply:
AHHHH I see, I didn't make the connection that `host` can be passed as a query parameter and that automatically hooks up things. When I saw parameter, I assumed it was a template parameter, instead of a query parameter passed in.

That's perfect, then I'll do as you say! Thanks again for clarification :).